src/main/kotlin/DirectoryLogic.kt

```diff
@@ -2,20 +2,57 @@ import java.lang.IllegalArgumentException
 
 class DirectoryLogic {
```

Comment:
DirectoryLogic just sounds like a giant miscellaneous drawer.
What directory? Directories in general? Is this utilities for working with relative and absolute paths? Sounds like if I want to find a function that lets me get a directory with respect to another directory, I'd try looking in here.

Also, why do I want to instantiate a DirectoryLogic class? It sounds like this class contains utilities. Object instantiation though implies state and side effects.

Now, I wonder which class I go to in order to catalog/index my files...